### PR TITLE
Code readability improvements

### PR DESCRIPTION
**Pull request by luciandavila**
Hello guys,

This PR is aiming to improve code readability by introducing the use of rubocop, a linter already used in other Barzahlen ruby repos. Since this is a customer-facing client, it is important that the code matches the community standards, being easily read.

Rubocop enforces many of the guidelines outlined in the community [Ruby Style Guide](https://rubystyle.guide/).  It can discover and report problems in the code, and can also automatically fix many of them.

The rubocop configuration proposed follows the [gcra-ruby repository configuration](https://github.com/Barzahlen/gcra-ruby/blob/master/.rubocop.yml), extending the company coding preferences to this repo. Some of the cops are still disabled, but if the maintainers approve the ideia, these could be activated in further changes.

A small change is also suggested in the rspec output format, making specs work as an extra documentation source. Just as a side note, this pull request might seem big, but most of the changes were automatically done by rubocop.

Kindly,
Lucian D'avila
Full Stack Developer

### CODE REVIEW

**Comment by tobischo** [comment]
Hello Lucian,
 I appreciate the effort. The rubocop config you added does not match the one we use at our company.

Granted, it did not either beforehand, but we do generally have a different set of preferences than the rubocop defaults. - Which we also currently do not apply on other projects for the simple reason of not having had the style defined when we made the last changes.

Therefore I will not merge your changes, but I will take the feedback about standardization to improve readability and see when we can introduce that for our open source projects.

File: .rspec [hunk 1]
```diff
@@ -1,2 +1,3 @@
 --color
 --require spec_helper
+--format documentation
```
Comment (tobischo) [inline .rspec:3]: Requires newline at the end of the file

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+
```
Comment (tobischo) [inline .rubocop.yml:1]: As mentioned, this is not our up to date config. You do not have access to that though, so this was pretty much an impossible task

**Comment by luciandavila** [comment]
@tobischo , thanks for the answer, it sounds perfectly reasonable.

> I will take the feedback about standardization to improve readability and see when we can introduce that for our open source projects.

It sounds very good to me, I'm glad to help!

**Pull request closed without merging**